crates/oxc_transformer/src/proposals/decorators.rs

```diff
@@ -391,10 +403,71 @@ impl<'a> Decorators<'a> {
+
                         def.decorators.clear();
+
+                        if def.key.is_private_identifier() {
+                            name = self.get_unique_name(&if def.computed {
+                                "init_computedKey".into()
+                            } else {
+                                format!("call_{}", def.key.name().unwrap()).into()
+                            });
+
+                            let mut arguments = self.ast.new_vec_with_capacity(2);
+                            arguments.push(Argument::Expression(self.ast.this_expression(SPAN)));
+                            arguments.push(Argument::Expression(
+                                self.ast.identifier_reference_expression(IdentifierReference::new(
+                                    SPAN,
+                                    "v".into(),
+                                )),
+                            ));
+                            def.value = self.ast.function(
+                                def.value.r#type,
+                                def.value.span,
+                                self.ast.copy(&def.value.id),
+                                def.value.generator,
+                                def.value.r#async,
+                                self.ast.copy(&def.value.this_param),
+                                self.ast.formal_parameters(
+                                    SPAN,
+                                    FormalParameterKind::FormalParameter,
+                                    self.ast.new_vec_single(self.ast.formal_parameter(
+                                        SPAN,
+                                        self.ast.binding_pattern(
+                                            self.ast.binding_pattern_identifier(
+                                                BindingIdentifier::new(SPAN, "v".into()),
+                                            ),
+                                            None,
+                                            false,
+                                        ),
+                                        None,
+                                        false,
+                                        self.ast.new_vec(),
+                                    )),
+                                    None,
+                                ),
+                                Some(self.ast.function_body(
+                                    SPAN,
+                                    self.ast.new_vec(),
+                                    self.ast.new_vec_single(
+                                        self.get_call_with_arguments(name.clone(), arguments),
+                                    ),
+                                )),
+                                self.ast.copy(&def.value.type_parameters),
+                                self.ast.copy(&def.value.return_type),
+                                self.ast.copy(&def.value.modifiers),
+                            );
```

Comment:
Here, we need to clone the function and modify the parameters and body. It seems that we need a clone function in `AstBuilder` if it is possible.

Reply:
Can we hack it somehow. Why is memcpy a nightly function lol https://doc.rust-lang.org/std/mem/fn.copy.html

Reply:
Looks like memcpy cannot solve. The copy function needs a copy trait. I thought we could add a `copy_function` function. This function implementation is like `move_expression`. create an empty function and replace it. What do you think?